lazy_tensor_core/lazy_tensor_core/csrc/ts_backend/aten_ltc_ts_type.cpp

```diff
@@ -250,6 +251,102 @@ at::Tensor LazyNativeFunctions::_copy_from_and_resize(const at::Tensor& self,
   return dst;
 }
 
+at::Tensor LazyNativeFunctions::_to_copy(const at::Tensor & self,
+                                         c10::optional<at::ScalarType> dtype,
+                                         c10::optional<at::Layout> layout,
+                                         c10::optional<at::Device> device,
+                                         c10::optional<bool> pin_memory,
+                                         bool non_blocking,
+                                         c10::optional<at::MemoryFormat> memory_format) {
+    
+    if (force_eager_fallback(at::aten::_to_copy)) {
+      TORCH_INTERNAL_ASSERT(false,
+        "Fallback is currently impossible for _to_copy since the fallback helper itself reinvokes _to_copy");
+    }
+
+    auto options = self.options();
+    if (dtype) {
+      // I put each of these setters in a conditional instead of doing `self.options().dtype(dtype).layout(layout)...
+      // because calling .dtype(nullopt) on an options() that already has dtype appears to wipe it
+      options = options.dtype(dtype);
+    }
+    if (layout) {
+      options = options.layout(layout);
+    }
+    // TODO(whc) can we honor 'pin_memory' in some/all cases?
```

Comment:
I think it makes sense in case 1 if we copy from a eager-cpu tensor to a lazy-cuda tensor?
Maybe in case 3 as well when we transfer to eager-cuda on a different device first?

I wonder if we should at least add a warning for case 1 and 3 so we can at least see when pinned memory is used

Reply:
yea, i think i could put a warning for now.  I am not sure if just respecting pinning would help/hurt in any cases.

Reply:
wait, why would case 2 not apply (if we go lazy->eager_cuda?)